pennylane/gradients/parameter_shift.py

```diff
@@ -1489,6 +1486,31 @@ def processing_fn(results):
                 combined_grad.append(meas_grad)
             return tuple(combined_grad)
 
+        # If there are unsupported parameters, we must process
+        # the quantum results separately, once for the fallback
+        # function and once for the parameter-shift rule, and recombine.
+        def processing_fn(results):
+            unsupported_grads = results[:fallback_len]
+            supported_grads = results[fallback_len:]
+
+            shot_vector = isinstance(shots, Sequence)
+            if not shot_vector:
+                unsupported_grads = fallback_proc_fn(unsupported_grads)
+                supported_grads = fn(supported_grads)
+                return _single_grad(unsupported_grads, supported_grads)
+
+            num_shot_vec_components = len(shots) if shot_vector else None

```

Comment:
I don't think `None` would ever be obtained here. Unless `shot_vector` is an empty sequence, but we don't allow that right?

Reply:
No, that's a good catch. Updated.